src/modules/storage/LedgerStorage.ts

```diff
@@ -1543,24 +1549,32 @@ export class LedgerStorage extends Storages {
                             }
                             default: {
                                 total_received = JSBI.BigInt(row[0].total_amount);
-                                if (Number(block.header.height) === 0) {
-                                    circulating_supply = total_received;
-                                }
                                 break;
                             }
                         }
-                        return this.query(circulating_supply_sql, [Number(block.header.height) - 1], conn);
+                        return this.query(get_utxo, [], conn);
                     })
-                    .then((row: any[]) => {
+                    .then(async (row: any[]) => {
                         if (row[0]) {
-                            circulating_supply = JSBI.add(JSBI.BigInt(row[0].circulating_supply), total_reward);
+                            total_utxo = JSBI.BigInt(row[0].total_utxo);
+                            let excludedAddresses: string[]
+                            if (this.excluded_addresses !== undefined) {
+                                excludedAddresses = this.excluded_addresses
+                            } else {
+                                excludedAddresses = []
+                            }
+                            await Promise.all(excludedAddresses.map(async (elem: string) => {
+                                const balance: any = await this.getWalletBalance(elem.toString());
+                                total_common_budget_balance = JSBI.add(total_common_budget_balance, JSBI.BigInt(balance[0].balance))
```

Comment:
`this.getWalletBalance` does not reflect data on new blocks because it is not the same transaction DB connection.

`balance[0].balance`  <- Index reference errors may occur.

Reply:
How about referring to it at the accounts table?

Reply:
If the balance of the accounts table of the corresponding block height has been calculated, it may be referred to.

Reply:
Or if you use 'this.getWalletBalance', you need to add a connection.

Reply:
I have added a connection,

Reply:
balance[0].balance <- Index reference errors may occur.

Reply:
check has been implement on balance[0]